diff_drive_controller/src/odometry.cpp

```diff
@@ -59,52 +58,49 @@ bool Odometry::update(double left_pos, double right_pos, const rclcpp::Time & ti
   const double right_wheel_cur_pos = right_pos * right_wheel_radius_;
 
   // Estimate velocity of wheels using old and current position:
-  const double left_wheel_est_vel = left_wheel_cur_pos - left_wheel_old_pos_;
-  const double right_wheel_est_vel = right_wheel_cur_pos - right_wheel_old_pos_;
+  const double left_wheel_est_vel = (left_wheel_cur_pos - left_wheel_old_pos_)/dt;
+  const double right_wheel_est_vel = (right_wheel_cur_pos - right_wheel_old_pos_)/dt;
 
   // Update old position with current:
   left_wheel_old_pos_ = left_wheel_cur_pos;
   right_wheel_old_pos_ = right_wheel_cur_pos;
 
-  updateFromVelocity(left_wheel_est_vel, right_wheel_est_vel, time);
+  updateFromVelocity(left_wheel_est_vel, right_wheel_est_vel, time, dt);
 
   return true;
 }
 
-bool Odometry::updateFromVelocity(double left_vel, double right_vel, const rclcpp::Time & time)
+bool Odometry::updateFromVelocity(double left_vel, double right_vel, const rclcpp::Time & time, const double dt)
 {
-  const double dt = time.seconds() - timestamp_.seconds();
-
   // Compute linear and angular diff:
   const double linear = (left_vel + right_vel) * 0.5;
   // Now there is a bug about scout angular velocity
   const double angular = (right_vel - left_vel) / wheel_separation_;
 
   // Integrate odometry:
-  integrateExact(linear, angular);
+  integrateExact(linear, angular, dt);
 
   timestamp_ = time;
```

Comment:
timestamp_ class variable could be removed now

Reply:
good catch